File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int N = matrix.size();
        int M = matrix[0].size();
        int l = 0, r = N * M - 1;
        while (l <= r) {
            int m = int((l + r) / 2);
            if (matrix[int(m / M)][m % M] == target) {
                return true;
            } else if (matrix[int(m / M)][m % M] > target) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        def dfs(i, comb, total):
            # 가지치기
            if total > target:
                return
            
            # 종료조건
            if total == target:
                answer.append(comb)
                return
            
            for nxt in range(i, len(nums)):
                dfs(nxt, comb + [nums[nxt]], total + nums[nxt])

            
        answer = []
        dfs(0, [], 0)
        return answer